### PR TITLE
`cargo install rerun-cli`

### PR DESCRIPTION
**Pull request by emilk**
Closes https://github.com/rerun-io/rerun/issues/1997
Closes https://github.com/rerun-io/rerun/pull/2177

Sibling PR:
* https://github.com/rerun-io/rerun-docs/pull/63

### What
`cargo install rerun-cli` installs the `rerun` binary.

This follows the pattern of e.g. the `wasm-bindgen-cli` crate which installs the `wasm-bindgen` binary, while there is also the `wasm-bindgen` library crate.

At the same time, `cargo add rerun` adds the `rerun` _library_ dependency, but WITHOUT the `native_viewer` and `web_viewer`  dependencies (the former one adds a lot of compile time). This is a great improvement for most users that are only using `rerun` as a logging library.

When developing, the preferred way of running the `rerun` binary is now with `cargo rerun`.

### Future work
We could move the whole `run.rs` part of `rerun` to `rerun-cli` and have `rerun_py` depend on the latter instead of the former. This would allow us to remove even more dependencies from `rerun`, improving its compile-time further

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)

PR Build Summary: https://build.rerun.io/pr/2183

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -17,6 +17,9 @@ This is a guide to how to build Rerun.
 * Run `./scripts/setup_dev.sh`.
 * Make sure `cargo --version` prints `1.69.0` once you are done
 
+You should now be able run our Rust examples, e.g. with `cargo run -p dna`.
```
Comment (Wumpf) [inline BUILD.md:20]: that sounds like a pr description not like a "how do I build things"

```suggestion
All Rust examples are separate executable projections. Meaning to run them you specify the package, e.g. `cargo run -p dna`.
```

File: crates/rerun-cli/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+[features]
+# The default is what the user gets when they call `cargo install rerun-cli`,
+# so wer have all the bells and wistles here
+default = ["native_viewer", "web_viewer"]
+
+## Support spawning a native viewer.
+## This adds a lot of extra dependencies, so only enable this feature if you need it!
+native_viewer = ["rerun/native_viewer"]
+
+## Support serving a web viewer over HTTP.
```
Comment (Wumpf) [inline crates/rerun-cli/Cargo.toml:32]: is there a deeper meaning to ## or is it a typo?

**Review by Wumpf — APPROVED** [review]
I like how this promotes a much faster way of working with rerun from rust (no native & webviewer, connect to installed)
it highlights though that we don't have an example illustrating that workflow though 🤔 

checks all out for me, let's do this :)

File: crates/rerun-cli/Cargo.toml (hunk 2 above)
Comment (emilk) [reply crates/rerun-cli/Cargo.toml:32]: There is a deeper meaning if one uses https://docs.rs/document-features/latest/document_features/ - which I should :)

**Pull request merged**